tests/components/unifiprotect/conftest.py

```diff
@@ -133,161 +110,227 @@ def mock_client(mock_bootstrap: Bootstrap):
 
-def assert_entity_counts(
-    hass: HomeAssistant, platform: Platform, total: int, enabled: int
-) -> None:
-    """Assert entity counts for a given platform."""
+    Sensor.__config__.validate_assignment = True
 
-    entity_registry = er.async_get(hass)
 
-    entities = [
-        e for e in entity_registry.entities if split_entity_id(e)[0] == platform.value
-    ]
+@pytest.fixture(name="sensor_all")
+def csensor_all_fixture(sensor: Sensor):
+    """Mock UniFi Protect Sensor device."""
 
-    assert len(entities) == total
-    assert len(hass.states.async_all(platform.value)) == enabled
+    all_sensor = sensor.copy()
+    all_sensor.light_settings.is_enabled = True
+    all_sensor.humidity_settings.is_enabled = True
+    all_sensor.temperature_settings.is_enabled = True
+    all_sensor.alarm_settings.is_enabled = True
+    all_sensor.led_settings.is_enabled = True
+    all_sensor.motion_settings.is_enabled = True
 
+    return all_sensor
 
-def ids_from_device_description(
-    platform: Platform,
-    device: ProtectAdoptableDeviceModel,
-    description: EntityDescription,
-) -> tuple[str, str]:
-    """Return expected unique_id and entity_id for a give platform/device/description combination."""
 
-    entity_name = (
-        device.name.lower().replace(":", "").replace(" ", "_").replace("-", "_")
-    )
-    description_entity_name = (
-        description.name.lower().replace(":", "").replace(" ", "_").replace("-", "_")
-    )
+@pytest.fixture(name="doorlock")
+def doorlock_fixture():
+    """Mock UniFi Protect Doorlock device."""
 
-    unique_id = f"{device.mac}_{description.key}"
-    entity_id = f"{platform.value}.{entity_name}_{description_entity_name}"
+    # disable pydantic validation so mocking can happen
+    Doorlock.__config__.validate_assignment = False
 
-    return unique_id, entity_id
+    data = json.loads(load_fixture("sample_doorlock.json", integration=DOMAIN))
+    yield Doorlock.from_unifi_dict(**data)
 
+    Doorlock.__config__.validate_assignment = True
 
-def generate_random_ids() -> tuple[str, str]:
-    """Generate random IDs for device."""
 
-    return random_hex(24).lower(), random_hex(12).upper()
+@pytest.fixture
+def unadopted_doorlock(doorlock: Doorlock):
+    """Mock UniFi Protect Light device (unadopted)."""
+
+    no_doorlock = doorlock.copy()
+    no_doorlock.name = "Unadopted Lock"
+    no_doorlock.is_adopted = False
+    return no_doorlock
 
 
-def regenerate_device_ids(device: ProtectAdoptableDeviceModel) -> None:
-    """Regenerate the IDs on UFP device."""
+@pytest.fixture
+def chime():
+    """Mock UniFi Protect Chime device."""
 
-    device.id, device.mac = generate_random_ids()
+    # disable pydantic validation so mocking can happen
+    Chime.__config__.validate_assignment = False
+
+    data = json.loads(load_fixture("sample_chime.json", integration=DOMAIN))
+    yield Chime.from_unifi_dict(**data)
+
+    Chime.__config__.validate_assignment = True
+
+
+@pytest.fixture(name="now")
```

Comment:
Maybe `fixed_now` ?

Reply:
I really do not want to rename all of the occurrences. Lol. Manually setting the name like this is just to make pylint happen since it complains about redefining the values.

Reply:
Not nearly as bad I thought it was going to be.